docs/develop/python/message-passing.mdx

```diff
@@ -27,155 +27,179 @@ tags:
+Here is an example Workflow Definition featuring Query, Signal, and Update handlers, and an Update validator.
+You can view this as a runnable sample at [hello/hello_message_passing.py](https://github.com/temporalio/samples-python/blob/dan/dan/docs_message_passing.py).
 
-- [Develop with Signals](#signals)
-- [Develop with Queries](#queries)
-- [Develop with Updates](#updates)
-- [Dynamic Handler](#dynamic-handler)
+```python
+import asyncio
+from dataclasses import dataclass
+from enum import IntEnum
+from typing import Optional
 
-## Signals {#signals}
+from temporalio import workflow
 
-**How to develop with Signals using the Python SDK.**
 
-A [Signal](/encyclopedia/workflow-message-passing#sending-signals) is a message sent asynchronously to a running Workflow Execution which can be used to change the state and control the flow of a Workflow Execution.
-It can only deliver data to a Workflow Execution that has not already closed.
+class Language(IntEnum):
+    Chinese = 1
+    English = 2
+    French = 3
+    Spanish = 4
+    Portuguese = 5
 
-Signals are defined in your code and handled in your Workflow Definition.
-Signals can be sent to Workflow Executions from a Temporal Client or from another Workflow Execution.
 
-There are two steps for adding support for a Signal to your Workflow code:
+@dataclass
+class GetLanguagesInput:
+    supported_only: bool
 
-1. **[Defining the Signal](#define-signal)** - You specify the name and data structure used by Temporal Clients when sending the Signal.
-2. **[Handling the Signal](#handle-signal)** - You write code that will be invoked when the Signal is received from a Temporal Client.
 
-After defining and handling your Signal, you can send it from a [Temporal Client](#send-signal-from-client) or from another [Workflow Execution](#send-signal-from-workflow).
+@dataclass
+class ApproveInput:
+    name: str
 
-### Define Signal {#define-signal}
 
-**How to define a Signal using the Python SDK.**
+@workflow.defn
+class GreetingWorkflow:
+    def __init__(self) -> None:
+        self.approved_for_release = False
+        self.approver_name: Optional[str] = None
+        self.language = Language.English
+        self.greetings = {
+            Language.English: "Hello, world",
+            Language.Chinese: "你好，世界",
+        }
 
-A Signal has a name and can have arguments.
+    @workflow.run
+    async def run(self) -> str:
+        await workflow.wait_condition(lambda: self.approved_for_release)
+        return self.greetings[self.language]
 
-- The name, also called a Signal type, is a string.
-- The arguments must be serializable.
-  To define a Signal, set the Signal decorator [`@workflow.signal`](https://python.temporal.io/temporalio.workflow.html#signal) on the Signal function inside your Workflow.
+    @workflow.query
+    def get_languages(self, input: GetLanguagesInput) -> list[Language]:
+        # A Query handler returns a value but must not mutate the Workflow state.
```

Comment:
Sure! Yeah, I was very focused on keeping sentence length down in those comments, since we want this code sample to appear "light weight".

Changing to

> A Query handler returns a value: it can inspect but must not mutate the Workflow state.

(Unfortunately, it _can_ mutate, so I prefer "must not")